test/general-tests/constructor-test/check.ts

```diff
@@ -1,8 +1,38 @@
-import { hardhatStarknetCompileDeprecated, hardhatStarknetTest } from "../../utils/cli-functions";
+import path from "path";
+import { hardhatStarknetCompile, hardhatStarknetTest } from "../../utils/cli-functions";
+import { copyFileSync } from "fs";
+import { assertContains, rmrfSync } from "../../utils/utils";
 
-hardhatStarknetCompileDeprecated(
-    "contracts/contract.cairo contracts/simple_storage.cairo contracts/empty_constructor.cairo".split(
-        " "
-    )
-);
+const prefix = path.join(__dirname);
+const sourcesPath = "cairo1-contracts";
+
+const duplicateConstructorContract = "duplicate_constructor.cairo";
+const contract1Path = path.join(sourcesPath, duplicateConstructorContract);
+
+const noConstructorContract = "no_constructor.cairo";
+const contract2Path = path.join(sourcesPath, noConstructorContract);
+
+const muteConstructorContract = "mute_constructor.cairo";
+const contract3Path = path.join(sourcesPath, muteConstructorContract);
+
+const commentedConstructorContract = "commented_constructor.cairo";
+const contract4Path = path.join(sourcesPath, commentedConstructorContract);
+
+const emptyLineConstructorContract = "empty_line_constructor.cairo";
+const contract5Path = path.join(sourcesPath, emptyLineConstructorContract);
+
+// Copy contracts to example repo to ensure files exists
+copyFileSync(path.join(prefix, duplicateConstructorContract), contract1Path);
+copyFileSync(path.join(prefix, noConstructorContract), contract2Path);
+copyFileSync(path.join(prefix, muteConstructorContract), contract3Path);
+copyFileSync(path.join(prefix, commentedConstructorContract), contract4Path);
+copyFileSync(path.join(prefix, emptyLineConstructorContract), contract5Path);
+
+const expectedErrorMsg = "Error: Expected at most one constructor.";
+const execution = hardhatStarknetCompile([contract1Path], true);
+assertContains(execution.stderr, expectedErrorMsg);
+rmrfSync(contract1Path);
+
+// Compile cairo1 contracts
+hardhatStarknetCompile(["cairo1-contracts/", "--add-pythonic-hints"]);
```

Comment:
Can't the sourcesPath variable be used here as well?

Reply:
https://github.com/0xSpaceShard/starknet-hardhat-plugin/pull/377/commits/81d8dd0f8431b12eac70fcdd226edc426bae3432